build.gradle

```diff
@@ -23,6 +23,50 @@ buildscript {
 
 plugins {
     id("io.gitlab.arturbosch.detekt").version("1.0.0-RC16")
+    id("com.nimbledroid.profiler").version("1.2.1")
+}
+
+// Placeholder for ND plugin configuration.
+//
+// The ND plugin expects this to be where gradle
+// stores configuration information. Variables
+// are set by the setNimbledroidConfiguration task.
+nimbledroid {
+}
+
+// Set the appropriate variables to perform the ND upload.
+//
+// Use the API key from the environment. Always upload a
+// nightly build.
+task setNimbledroidConfiguration {
+    doLast {
+        if (System.env.ND_APIKEY == null) {
```

Comment:
fwiw, this doesn't follow the convention we already have for adding API keys: https://searchfox.org/mozilla-mobile/rev/574cb270fae8924647990be2be04ca223774438e/fenix/app/build.gradle#267

I'm not sure it's the best practice (e.g. it should be built into a re-usable function too!) but it's good to be consistent so that it's easier to change later (i.e. refactor them all to one function and then change the implementation of that function).

Reply:
Definitely see your point here, but I'm wondering if there is a difference that is important: those things stored in files seem to be described as tokens and not keys. Are tokens supposed to be secret? It seems to me that we want to keep keys in environment variables (a best practice) rather than putting them anywhere on disk. 

What do you think?

Reply:
Those files are supposed to be kept secret: they're added in the CI environment and for local developers who need access to them. Maybe files are more convenient for local developers, especially those who may not know the best practices for setting up environment variables.

In any case, I'm not too opinionated here but I don't consider this key much different than the other ones so I'd lean towards consistency but I haven't thought through the implications.